applicationset/generators/cluster_profile.go

```diff
@@ -0,0 +1,251 @@
+package generators
```

Comment:
this file still seems to have a lot of references to secrets, which may be because you forked it from the Secrets-based generator. I'm guessing it's still a WIP, could you update us when it's ready?

If it's ready for review, Ill have a closer look but it sounds like there may be a few references to secrets we could get rid of